### ISSUE TITLE
nyc instrument resolves source maps incorrectly

### ISSUE THREAD

**Issue by zoehneto** [istanbuljs/nyc#793]
## Expected Behavior
When running `nyc instrument ./inputFolder ./outputFolder` containing javascript files with a reference to a sourcemap in the same directory like `//# sourceMappingURL=filename.js.map`, nyc should find this sourcemap. This format is used by both tsc and the rollup sourcemap plugin.

## Observed Behavior
nyc can't find the sourcemap because it searches for it in the parent directory: 
```
An error occurred while trying to read the map file at /home/user/parentDirectory/filename.js.map
Error: ENOENT: no such file or directory, open '/home/user/parentDirectory/filename.js.map'
```

## Forensic Information

**Operating System:**
Arch Linux 64bit
**Environment Information:** 
node v9.8.0
npm 5.7.1
nyc 11.6.0

**Comment by tex0l** [comment]
I see the same thing

**Comment by zoehneto** [comment]
As a workaround you can change into the directory with the files to be instrumented and execute `nyc instrument . .` there.

**Comment by Swatinem** [comment]
Hm, I have a related issue, which is that files with inline sourcemaps are wrong.
Instanbul just searches for a file on the filesystem with the name `/path/to/file/data:application/json;charset=utf-8;base64,XXXXX`

**Comment by JaKXz** [comment]
Hopefully this will be resolved in #893.
